server/src/packages/rwcds-form/rwcds-form.service.ts

```diff
@@ -171,14 +176,30 @@ export class RwcdsFormService {
       const projectActionLabel = action['_dcp_action_value@OData.Community.Display.V1.FormattedValue'];
 
       // Lookup: dcp_affectedzoningresolutions asks for a ZR type,
-      const { label, code } = ZONING_RESOLUTION_TYPES.filter((zr) => zr.label === projectActionLabel)[0] || { label: '', code: null };
+      const { label, code } = ZONING_RESOLUTION_TYPES.find((zr) => zr.label === projectActionLabel) || { label: '', code: null };
+      const matchingZr = affectedZoningResolutions.find(zr => zr.dcp_zoningresolutiontype === code);
+      // action attribute that matches dcp_zrsectionnumber on dcp_affectedzoningresolution
+      const currentActionZoningResolution = action.dcp_ZoningResolution ? action.dcp_ZoningResolution.dcp_zoningresolution : null
 
       if (!label) console.log(`Could not find Affected ZR Type for ${projectActionLabel}`);
 
-      if (!zrTypes.includes(label) && label) {
+      // IF the action.ZoningResolution.dcp_zoningresolution value has changed since the initial POST,
+      // then PATCH the new value to dcp_zrsectionnumber on dcp_affectedzoningresolution entity
+      // NOTE: dcp_modifiedzrsectionnumber not included here because it's updated by applicant on rwcds-form
+      if (matchingZr && matchingZr.dcp_zrsectionnumber !== currentActionZoningResolution) {
+        return this.crmService.update('dcp_affectedzoningresolutions', matchingZr.dcp_affectedzoningresolutionid, {
+          dcp_zrsectionnumber: currentActionZoningResolution,
+        });
+      }
+
+      // IF the zr type does not exist yet, then POST the action
+      // NOTE: some actions that exist in dcp_projectaction SHOULD NOT be copied over
+      // NOTE cont. -- The ZONING_RESOLUTION_TYPES lookup also functions as a list of actions that SHOULD...
+      // NOTE cont. -- be copied over from dcp_projectaction, which is why we check that `label` exists here.
+      if (!matchingZr && label) {
```

Comment:
If matchingZr doesn't exist, why does it matter that label exist? 
It seems like more important there should be a check that `code` exists

Reply:
It doesn't matter whether it's `code` vs `label`. What DOES matter is that the action exists in the ZONING_RESOLUTION_TYPES. This lookup also functions as a list of actions that SHOULD be copied over. For example, every rwcds-form requires an `EAS` action but this action should never be copied over to the dcp_affectedzoningresolution entity. So when looping through the dcp_projectactions entity, the EAS action would produce an undefined label/code. When we don't have that extra label/code check here, the function will POST an extra row with a null `dcpZoningresolutiontype` to dcp_affectedzoningresolution (because it would be POSTing this EAS action). In order to keep this from happening, we need to make sure that code/label exists

Reply:
Ah gotcha, that makes sense, thank you for clarifying. Sorry that's my bad for missing it in your comment above. I think `code` will be more clear here, since ultimately that is posted to the new affected zoning resolution.

Reply:
You're all good! Sounds good, I'll change it to `code` :)